src/Features/Core/Portable/AddPackage/InstallPackageDirectlyCodeActionOperation.cs

```diff
@@ -67,12 +69,13 @@ internal class InstallPackageDirectlyCodeActionOperation : CodeActionOperation
 
         internal override bool ApplyDuringTests => true;
 
-        internal override bool TryApply(
+        internal override Task<bool> TryApplyAsync(
             Workspace workspace, IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            return _installerService.TryInstallPackage(
+            var result = _installerService.TryInstallPackage(
                 workspace, _document.Id, _source, _packageName,
                 _versionOpt, _includePrerelease, cancellationToken);
+            return result ? SpecializedTasks.True : SpecializedTasks.False;
```

Comment:
this will become async end to end once this and https://github.com/dotnet/roslyn/pull/55671 goes in.

Reply:
i stitch it together in https://github.com/dotnet/roslyn/pull/55759 :)